### PR TITLE
[mod_articles_category] Grouping by specified date field.

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue #19661.

### Summary of Changes

Currently the module groups articles only by created date. This PR adds an option to choose which date field (created, modified or publish_up) to use for grouping articles.

### Testing Instructions

Create 'Articles - Category' module.
Enable grouping by Year or by Month and Year.
Select a date field to group by.

### Expected result

Grouping works.

### Documentation Changes Required
Help page needs updating https://help.joomla.org/proxy?keyref=Help38:Extensions_Module_Manager_Articles_Category&lang=en
Also docblock was updated, IDK if that matters.

### CODE REVIEW

File: modules/mod_articles_category/mod_articles_category.xml [hunk 1]
```diff
@@ -348,6 +348,28 @@
...
 
+				<field
+					name="month_year_format"
+					type="text"
+					label="MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_LABEL"
+					description="MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_DESC"
+					default="F Y"
+					showon="article_grouping:year,month_year"
+				/>
+
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.xml]: Remove tabs

File: modules/mod_articles_category/mod_articles_category.xml [hunk 2]
```diff
@@ -348,6 +348,28 @@
...
+					label="MOD_ARTICLES_CATEGORY_FIELD_DATEGROUPINGFIELD_LABEL"
+					description="MOD_ARTICLES_CATEGORY_FIELD_DATEGROUPINGFIELD_DESC"
+					default="created"
+					showon="article_grouping:year,month_year"
+					>
+					<option value="created">MOD_ARTICLES_CATEGORY_OPTION_CREATED_VALUE</option>
+					<option value="modified">MOD_ARTICLES_CATEGORY_OPTION_MODIFIED_VALUE</option>
+					<option value="publish_up">MOD_ARTICLES_CATEGORY_OPTION_STARTPUBLISHING_VALUE</option>
+				</field>
+
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.xml]: Remove tabs

File: modules/mod_articles_category/mod_articles_category.xml [hunk 3]
```diff
@@ -358,14 +380,7 @@
...
 				</field>
-
-				<field
-					name="month_year_format"
-					type="text"
-					label="MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_LABEL"
-					description="MOD_ARTICLES_CATEGORY_FIELD_MONTHYEARFORMAT_DESC"
-					default="F Y"
-				/>
+
```
Comment (Quy) [inline modules/mod_articles_category/mod_articles_category.xml]: Remove tabs

**Comment by SharkyKZ** [comment]
Updated.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c53995cc26069191419f77c785b70cff4ecc9d61<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20139">issues.joomla.org/tracker/joomla-cms/20139</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on c53995cc26069191419f77c785b70cff4ecc9d61<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20139">issues.joomla.org/tracker/joomla-cms/20139</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by SharkyKZ** [comment]
Closing this in favor of #20248.

**Pull request closed without merging**